src/helpers/purchase-operation-helper.ts

```diff
@@ -195,6 +197,15 @@ export class PurchaseOperationHelper {
                   });
                   return;
                 case CheckoutSessionStatus.Failed:
+                  const errorCode = operationResponse.operation.error?.code;
+                  if (
+                    errorCode ===
+                    CheckoutStatusErrorCodes.SetupIntentCompletionFailed
```

Comment:
If this is the error we'll receive when Stripe detected errors on their side maybe we shouldn't call `scheduleNextAttempt` since the user will be interacting again with the form.

Reply:
Sure, on the backend, we mark the `operation session` as failed when the `setup intent` fails to complete. However, according to Stripe’s docs, it’s still possible to [complete the setup intent](https://docs.stripe.com/payments/setupintents/lifecycle).

To address this, I updated the SDK to treat this error code as if the `setup_intent` was still `in_progress`. This is safe because we only switch to the `polling-purchase-status` state after successfully completing the `setup_intent`.

This ensures backward compatibility with previous SDK versions, as the backend behavior remains unchanged.

Reply:
Yeah but why do we keep trying.
Don't we have to wait for the user to click "Confirm" or "Pay" again?

When the user clicks the button again it will restart the attempts and we risk to have 2 timeout running at the same time.

Reply:
We agreed to handle this on the backend, so `operation sessions` will not be marked as `failed` when `setup intents` fail to complete.